lib/mosaic-geomath/src/main/java/org/eclipse/mosaic/lib/geo/CartesianPolygon.java

```diff
@@ -90,6 +89,40 @@ public boolean contains(CartesianPoint point) {
         return MathUtils.pnpoly(vertices.size(), verticesXValues, verticesYValues, (float) point.getX(), (float) point.getY());
     }
 
+    /**
+     * Returns true if there is a collision with another polygon.
+     *
+     * @param polygon The other polygon
+     * @return true if the polygons collide
+     */
+    public boolean isIntersectingPolygon(CartesianPolygon polygon) {
+        // Test if bounding boxes intersect
+        CartesianRectangle rectA = polygon.boundingBox;
+        CartesianRectangle rectB = calcBoundingBox(vertices);
+        if ((rectA.getA().getX() > rectB.getB().getX() || rectB.getB().getX() < rectB.getA().getX()
```

Comment:
you could also move this to a new method so that you can call `rectA.intersects(rectB) || rectB.intersects(rectA)`

Reply:
It would then be just rectA.intersects(rectB). As in one call the method would test whether rectA is ocmpletely above, below, left or right from rectB. No need then to check it in the other direction. So a new method doesn't make sense I think.

Reply:
I just had readability in mind ;) `if (rectA.intersects(rectB)) ` reads much better than how it is currently implemented.

Reply:
Ok, it's changed